clang/test/Driver/sycl-offload-intelfpga.cpp

```diff
@@ -29,6 +29,22 @@
 // CHK-RANGE-ROUNDING: clang{{.*}} "-fsycl-is-device"{{.*}} "-fsycl-disable-range-rounding"
 // CHK-RANGE-ROUNDING: clang{{.*}} "-fsycl-disable-range-rounding"{{.*}} "-fsycl-is-host"
 
+/// FPGA target implies -emit-non-kernel-entry-points=0 in sycl-post-link
+// RUN:   %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fintelfpga %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-NON-KERNEL-ENTRY-POINTS %s
+// RUN:   %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_fpga-unknown-unknown %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-NON-KERNEL-ENTRY-POINTS %s
+// CHK-NON-KERNEL-ENTRY-POINTS: sycl-post-link{{.*}} "-emit-non-kernel-entry-points=0"
+
+/// Non-FPGA targets should not imply -emit-non-kernel-entry-points=0 in sycl-post-link
+// RUN:   %clang -### -fsycl -fsycl-targets=spir64_fpga,spir64_gen %s 2>&1 \
+// RUN:   | FileCheck %s --check-prefix=CHK-NON-KERNEL-ENTRY-POINTS-NEG-1
+// CHK-NON-KERNEL-ENTRY-POINTS-NEG-1: sycl-post-link{{.*}} "-emit-non-kernel-entry-points=0"
```

Comment:
I'd also suggest checking that correct FE calls are being made prior to the expected post-link calls (`clang -cc1 -fsycl-is-device {{.*}} -triple spir64_fpga-unknown-unknown` before the first sycl-post-link check, `spir64_gen` before the second). Should the sorting of target toolchains that takes place internally change at some point, a failure in this test would be easier to debug that way.

Reply:
Good idea, added in b88f2c5